src/cart/crt_context.c

```diff
@@ -221,13 +221,15 @@ crt_context_provider_create(crt_context_t *crt_ctx, int provider)
 		D_GOTO(out, rc);
 	}
 
-	rc = crt_hg_get_addr(ctx->cc_hg_ctx.chc_hgcla,
-			     ctx->cc_self_uri, &uri_len);
-	if (rc != 0) {
-		D_ERROR("ctx_hg_get_addr() failed; rc: %d.\n", rc);
-		D_RWLOCK_UNLOCK(&crt_gdata.cg_rwlock);
-		crt_context_destroy(ctx, true);
-		D_GOTO(out, rc);
+	if (crt_is_service() || !crt_na_type_is_ucx(provider)) {
+		rc = crt_hg_get_addr(ctx->cc_hg_ctx.chc_hgcla,
+				     ctx->cc_self_uri, &uri_len);
```

Comment:
how is self_uri set for ucx?
crt_self_uri_get() and crt_rank_uri_get() APIs rely on this value to be set; this API is then used in number of tests